### ISSUE TITLE
Debugger: Game hangs when hovering over initialized constructors using static_get() to see their values

### ISSUE THREAD

**Issue by knno** [YoYoGames/GameMaker-Bugs#4884]
### Description

Writing this code in a script  causes an issue with the debugger behavior and "might" lead to issues with VM builds.

```gml
function Thing() constructor {
  static opts = [1,2,3]
}
Thing();

// This causes an issue
variable_global_set("Thing", Thing); // or (static_get(Thing))
```
And
```gml
// Some object creation code and the instance is in room
my_thing = static_get(Thing); // buggy
my_thing = Thing; // ref script
````
Then, during debugging, the issue occurs here:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/15669932/662d06e7-f47d-4a98-89e9-1613b88a0518)
and results in:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/15669932/7257be20-68e1-437a-80ff-321b6e5a7607)

[Project1.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14873459/Project1.zip)

### Steps To Reproduce

1. Start GameMaker
2. Import the attached project
3. Open the object and put a breakpoint on line 1 of its Create event
4. Debug the game (and allow any firewall prompts) 
5. It will immediately hit the breakpoint
6. Click focus back onto GM, rather than the runner
7. Mouse over "hi" at the start of the line
8. Observe the IDE immediately changes to signify the game connection has been lost
9. Observe the runner is also in a hang state and you have to close it manually

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.2.0 (Monthly)

### Which operating system(s) are you seeing the problem on?

Windows 11

### Which platform(s) are you having trouble building this project for?

Windows

### Are you running GameMaker from inside your Steam library?

No

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by YYDan** [comment]
Confirmed in Feb monthly and also still the case in IDE v2024.400.0.533  Runtime v2024.400.0.552.

Cleaned up the sample and also expanded the repro steps.

**Comment by rwkay** [comment]
Fixed in 2024.4

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.400.0.539  Runtime v2024.400.0.558
